python/pyarrow/meson.build

```diff
@@ -0,0 +1,398 @@
+        subdir: 'pyarrow',
+    )
+
+    install_data(cython_mod_name, install_dir: py.get_install_dir() / 'pyarrow')
+endforeach
+
+cython_headers = [
+    '_acero.pxd',
+    'array.pxi',
+    'benchmark.pxi',
+    'builder.pxi',
+    'compat.pxi',
+    '_compute.pxd',
+    'config.pxi',
+    '_csv.pxd',
+    '_cuda.pxd',
+    '_dataset_parquet.pxd',
+    '_dataset.pxd',
+    'device.pxi',
+    '_dlpack.pxi',
+    'error.pxi',
+    '_fs.pxd',
+    '__init__.pxd',
+    'io.pxi',
+    'ipc.pxi',
+    '_json.pxd',
+    'lib.pxd',
+    'memory.pxi',
+    '_orc.pxd',
+    'pandas-shim.pxi',
+    '_parquet_encryption.pxd',
+    '_parquet.pxd',
+    'public-api.pxi',
+    '_pyarrow_cpp_tests.pxd',
+    'scalar.pxi',
+    'table.pxi',
+    'tensor.pxi',
+    'types.pxi',
+]
+
+install_data(cython_headers, install_dir: py.get_install_dir() / 'pyarrow')
+
+install_subdir('includes', install_dir: py.get_install_dir() / 'pyarrow')
+
+install_subdir(
+    'src/arrow/python',
+    install_dir: py.get_install_dir() / 'pyarrow/src/arrow',
+)
+
+pysources = [
+    'acero.py',
+    'benchmark.py',
+    'cffi.py',
+    '_compute_docstrings.py',
+    'compute.py',
+    'conftest.py',
+    'csv.py',
+    'cuda.py',
+    'dataset.py',
+    'feather.py',
+    'flight.py',
+    'fs.py',
+    '_generated_version.py',
+    '__init__.py',
+    'ipc.py',
+    'json.py',
+    'jvm.py',
+    'orc.py',
+    'pandas_compat.py',
+    'substrait.py',
+    'types.py',
+    'util.py',
+]
+py.install_sources(pysources, subdir: 'pyarrow')
+
+py.install_sources(
+    files(
+        'src/arrow/python/api.h',
+        'src/arrow/python/arrow_to_pandas.h',
+        'src/arrow/python/async.h',
+        'src/arrow/python/benchmark.h',
+        'src/arrow/python/common.h',
+        'src/arrow/python/csv.h',
+        'src/arrow/python/datetime.h',
+        'src/arrow/python/decimal.h',
+        'src/arrow/python/extension_type.h',
+        'src/arrow/python/filesystem.h',
+        'src/arrow/python/flight.h',
+        'src/arrow/python/gdb.h',
+        'src/arrow/python/helpers.h',
+        'src/arrow/python/inference.h',
+        'src/arrow/python/io.h',
+        'src/arrow/python/ipc.h',
+        'src/arrow/python/iterators.h',
+        'src/arrow/python/numpy_convert.h',
+        'src/arrow/python/numpy_init.h',
+        'src/arrow/python/numpy_interop.h',
+        'src/arrow/python/numpy_to_arrow.h',
+        'src/arrow/python/parquet_encryption.h',
+        'src/arrow/python/pch.h',
+        'src/arrow/python/platform.h',
+        'src/arrow/python/pyarrow.h',
+        'src/arrow/python/pyarrow_lib.h',
+        'src/arrow/python/python_test.h',
+        'src/arrow/python/python_to_arrow.h',
+        'src/arrow/python/type_traits.h',
+        'src/arrow/python/udf.h',
+        'src/arrow/python/visibility.h',
+    ),
+    subdir: 'pyarrow/include/arrow/python',
+)
+
+if arrow_dep.found()
+    arrow_header_dir = arrow_dep.get_variable(
+        cmake: 'ARROW_INCLUDE_DIR',
```

Comment:
In Crossbow I see this is failing in an environment where cmake is present but not pkgconfig. It looks like `ARROW_INCLUDE_DIR` is not an exported variable, but rather a local variable pointing to the `INTERFACE_INCLUDE_DIRECTORIES` property of the `arrow_shared` target.

I couldn't find a good way in Meson to extract that property from a target, if its even possible. Maybe the CMake configuration needs to export this as a variable so other build systems can benefit?